packages/relay/src/lib/eth.ts

```diff
@@ -580,21 +582,20 @@ export class EthImpl implements Eth {
       }
     }
 
-    // check cache first
-    // create a key for the cache
-    const cacheKey = `${constants.CACHE_KEY.ETH_GET_BALANCE}-${account}-${blockNumberOrTag}`;
-    const cachedBalance = this.cache.get(cacheKey);
-    if (cachedBalance) {
-      this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(cachedBalance)}`);
-      return cachedBalance;
-    }
-
     let blockNumber = null;
     let balanceFound = false;
     let weibars: BigInt = BigInt(0);
     const mirrorAccount = await this.mirrorNodeClient.getAccount(account, requestId);
 
     try {
+      // check balance cache first for high load scenarios
+      const cacheKey = `${constants.CACHE_KEY.ETH_GET_BALANCE}-${account}-${blockNumberOrTag}`;
+      const cachedBalance = this.cache.get(cacheKey);
+      if (cachedBalance) {
+        this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(cachedBalance)}`);
+        return cachedBalance;
+      }
+
       if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
         const block = await this.getHistoricalBlockResponse(blockNumberOrTag, true, requestId);
```

Comment:
This is a dupe of the blockNumber() call before and will go get the new latest except it doesn't consider a cache.
Additionally there's some buggy logic in there for earliest tag also.
If we're going to restore the logic for historical calculations we should still do a perf check and optimize it

Reply:
Moved the cache up and removed the dup so the cache is checked.